### ISSUE TITLE
(Re)move Debian packages in `packages.list` and `packages_minimal.list`

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#6566]
What's the purpose of these files?

Why are some packages installed there?

* https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages.list
* https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages_minimal.list

full list:

```
ncurses-term
aptitude
tasksel
sudo
dmsetup
psmisc
gnupg
xterm
libfile-mimeinfo-perl
libglib2.0-bin
ltrace
strace
haveged
wireless-tools
wpasupplicant
dbus-x11
```

For example, having `wpasupplicant` there seems most obviously unnecessary. That results in the Whonix template having this package, although not required.

Wouldn't it be better if these packages where referenced in [qubes meta packages](https://github.com/QubesOS/qubes-meta-packages/blob/master/debian/control)?

**Comment by unman** [comment]
We could move those to "vm-recommended" where the networking stuff is included.
I dont think the rest are candidates for removal, but I'll check.

**Comment by 3hhh** [comment]
`haveged` nowadays only makes sense for older kernels as newer ones do its job inside the kernel.

**Comment by TNTBOMBOM** [comment]
#8330 related.

**Comment by marmarek** [comment]
Wireless tools removed here: https://github.com/QubesOS/qubes-builder-debian/pull/74

**Comment by TNTBOMBOM** [comment]
PR: Remove strace and lstrace

https://github.com/QubesOS/qubes-builder-debian/pull/75

**Comment by TNTBOMBOM** [comment]
PR: Remove haveged and dbus-x11

https://github.com/QubesOS/qubes-builder-debian/pull/77

**Comment by unman** [comment]
I do not think haveged is unnecessary - it still has a place.
dbus-x11 is marginal, but is of use for KDE.
What is the (detailed) reason for wanting to remove them?

**Comment by TNTBOMBOM** [comment]
> I do not think haveged is unnecessary - it still has a place.

Needs evidence to be proven.

> dbus-x11 is marginal, but is of use for KDE.

KDE is Wayland now, but nevertheless, it still requires evidence to be proven.

**Comment by adrelanos** [comment]
> The trace packages remain useful for troubleshooting, and have marginal impact. What is the reason for removing them?

The idea of this ticket is to remove these packages from [packages_minimal.list](https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages_minimal.list) and if worthwhile, properly reintroduce these by adding these to a Qubes meta packages (qubes-meta-packages).

> haveged is still useful for entropy with old kernels (e.g. some in-VM kernels). Admittedly these are uncommon configurations for Qubes OS, but they may happen.

Which distributions? In other words, how long do we have to carry this legacy?

> It may be better to keep it and install a systemd condition to not start the daemon on newer kernels. The haveged author proposed just that a long time ago. Possibly it even made its way into debian by now - I haven't checked.

Seems a lot effort just to support some outdated (possibly already deprecated?) templates.

> dbus-x11 is marginal, but is of use for KDE.

What's the benefit for dbus-x11?

If dbus-x11 is required, let's properly add it to a Qubes meta package instead?

**Comment by andrewdavidwong** [comment]
Why are people arguing about which non-Qubes packages are valuable, useful, or desirable for a Debian template? Isn't the whole point of [our policy about upstream distros](https://www.qubes-os.org/faq/#what-is-qubes-attitude-toward-changing-guest-distros) to avoid *precisely* these sorts of arguments?

**Comment by marmarek** [comment]
The minimal template intentionally deviate from this policy, to be as small base as possible. But this goes only one way - if a package isn't included in default package set of a distribution, surely it shouldn't be included in the minimal template either (unless needed by something qubes-specific, ofc).

**Comment by andrewdavidwong** [comment]
> The minimal template intentionally deviate from this policy, to be as small base as possible.

Yes, but in that case, it's still irrelevant whether a package is valuable, useful, or desirable. The standard for minimal templates isn't "be as small as possible *while including all of the valuable, useful, and desirable stuff*." It's just "be as small as possible."

In this case, "possible" really means something more like "practical" or "feasible." In other words, the minimal templates aim to be as small as possible *while still serving their function as templates*. So, people arguing over whether packages are valuable, useful, or desirable in the context of minimal templates are having the wrong debate. They must instead argue that the template *will not work* or *will not be able to serve its purpose* without the package in question, which is a much higher bar.

**Comment by TNTBOMBOM** [comment]
Thank you @marmarek @andrewdavidwong. 

https://github.com/QubesOS/qubes-issues/issues/6566#issuecomment-1647610923 can we proceed so we i can post the next PR?

**Comment by unman** [comment]
On Tue, Jul 23, 2024 at 03:23:29PM -0700, duck09 wrote:
> What about the packages listed in this forum [post](https://forum.qubes-os.org/t/minifying-debian-12-minimal/24778)?

You'll see that there was a discussion (brief) in qubes-devel, but the
conclusion was that we should honor the distro in producing minimal
templates.
While it is possible to produce smaller templates (either by rebuilding
with small package list or by package removal), these do not provide
what Debian determines is the core of Debian. The same applies to
Fedora.

**Comment by adrelanos** [comment]
May I suggest to limit the scope of this ticket?

The scope being these two source code files only:

* https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages.list
* https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages_minimal.list

Packages listed there either being removed or moved to [qubes meta packages](https://github.com/QubesOS/qubes-meta-packages/blob/master/debian/control).

This would be specifically useful in case of `packages_minimal.list` cleanup. Because then other distributions Templates built using `qubes-builder-debian` can have less packages installed by default.

If a package is not listed in these files, I would appreciate if that could be discussed in a different ticket, mailing list discussion or forum topic.

**Comment by TNTBOMBOM** [comment]
https://github.com/QubesOS/qubes-builder-debian/pull/83

**Comment by TNTBOMBOM** [comment]
PR remove xterm and libfile-mimeinfo-perl:

https://github.com/QubesOS/qubes-builder-debian/pull/84

https://github.com/QubesOS/qubes-builder-debian/pull/84/commits/3724896d2a30a70d24307c4a584613af5ea5ee35

**Comment by TNTBOMBOM** [comment]
> Is the inclusion of `gnupg` and `libglib2.0-bin` necessary?

https://github.com/QubesOS/qubes-builder-debian/pull/85

**Comment by TNTBOMBOM** [comment]
> Do you know if only the Gateway is based on the minimal template or is the Workstation based on it too?

We have a section in our [forum](https://forums.whonix.org/) you can ask there.

**Comment by ben-grande** [comment]
@adrelanos Anything else to do?

Current minimal list is seems enough: https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages_minimal.list

Current full list has something extra?: https://github.com/QubesOS/qubes-builder-debian/blob/master/template_debian/packages.list

**Comment by adrelanos** [comment]
Minimal is good enough. The full list is a matter of taste.
